src/std/sxml/html/parser.ss

```diff
@@ -0,0 +1,74 @@
+(import :gerbil/gambit :std/error :std/srfi/1 :std/srfi/13 :std/text/utf8
+  :std/sxml/print)
+(export
+  html->sxml
+  default-html->sxml-plist
+  html-strip
+  make-html-parser
+
+  sxml->html
+  html-escape
+  html-character-escapes)
+
+(def (integer->utf8-string n) (utf8->string (u8vector n)))
+
+(def (print-sxml sxml (out (current-output-port)))
+  (sxml-display-as-html sxml out))
+
+(def (sxml->html sxml (port #f))
+  (if port (print-sxml sxml port) (%sxml->html sxml)))
+(def (html-escape str (port #f) escapes: (esc #f))
+  (unless esc (set! esc html-character-escapes))
+ (if port (html-display-escaped-string str port esc) (%html-escape str esc)))
+
+(def default-html->sxml-plist
+  [start: (lambda (tag attrs seed virtual?) '())
+   end:   (lambda (tag attrs parent-seed seed virtual?)
+            `((,tag ,@(if (pair? attrs)
+                        `((@ ,@attrs) ,@(reverse seed))
+                        (reverse seed)))
+              ,@parent-seed))
+   decl:    (lambda (tag attrs seed) `((*DECL* ,tag ,@attrs) ,@seed))
+   process: (lambda (attrs seed) `((*PI* ,@attrs) ,@seed))
+   comment: (lambda (text seed) `((*COMMENT* ,text) ,@seed))
+   text:    (lambda (text seed) (cons text seed))])
```

Comment:
this should be a hash table, I don't see any benefit in using a plist (and it is slow to read)

Reply:
Agreed. It was at one point a plist that was applied and I did not change it when the lambda list changed.